warehouse/macaroons/services.py

```diff
@@ -140,6 +140,16 @@ def verify(self, raw_macaroon: str, request, context, permission) -> bool:
         if dm is None:
             raise InvalidMacaroonError("deleted or nonexistent macaroon")
 
+        # Originally our Macaroons did not have a caveat to indicate what permissions
+        # they were valid for, however they now do (and we currently always emit them
+        # with a permission caveat), so if this particular Macaroon predates that then
+        # we will do a legacy fallback here.
+        if dm.predates_permission_caveat:
+            if permission != "upload":
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Would this also need to be replaced by an enum?